src/main/java/core/basesyntax/dao/figure/FigureDaoImpl.java

```diff
@@ -12,11 +17,37 @@ public FigureDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public T save(T figure) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(figure);
+            transaction.commit();
+            return figure;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save figure: " + figure + " to db");
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<T> findByColor(String color, Class<T> clazz) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<T> query = criteriaBuilder.createQuery(clazz);
+            Root<T> root = query.from(clazz);
+            query.where(criteriaBuilder.equal(root.get("color"), color));
+            return session.createQuery(query).getResultList();
```

Comment:
`JFYI` we should use criteria API for dynamic queries. In this case it's a static query so it's better to use HQL here